packages/core/src/Gateway.ts

```diff
@@ -0,0 +1,12 @@
+import type { Awaitable } from '@discordjs/util';
+import type { ManagerShardEventsMap, WebSocketShardEvents } from '@discordjs/ws';
+import type { GatewaySendPayload } from 'discord-api-types/v10';
+
+export interface Gateway {
+	getShardCount(): Awaitable<number>;
+	on(
+		event: WebSocketShardEvents.Dispatch,
+		listener: (...params: ManagerShardEventsMap[WebSocketShardEvents.Dispatch]) => Awaitable<void>,
+	): this;
```

Comment:
Does this use only `WebSocketShardEvents.Dispatch` for a reason? Since this makes the typings incomplete for other events.

Reply:
Yes, `/core` needs only dispatch

Reply:
Understandable, thanks.